newIDE/app/src/EventsSheet/EventsTree/index.js

```diff
@@ -128,80 +130,92 @@ type EventsContainerProps = {|
  * It will report the rendered event height so that the EventsTree can
  * update accordingly.
  */
-class EventContainer extends Component<EventsContainerProps, {||}> {
-  _container: ?any;
-  componentDidMount() {
-    const height = this._container ? this._container.offsetHeight : 0;
-    this.props.eventsHeightsCache.setEventHeight(this.props.event, height);
-  }
-
-  componentDidUpdate() {
-    const height = this._container ? this._container.offsetHeight : 0;
-    this.props.eventsHeightsCache.setEventHeight(this.props.event, height);
-  }
+const EventContainer = (props: EventsContainerProps) => {
+  const {
+    event,
+    project,
+    scope,
+    disabled,
+    eventsHeightsCache,
+    onEventContextMenu,
+  } = props;
+  const forceUpdate = useForceUpdate();
+  const containerRef = React.useRef<?HTMLDivElement>(null);
+  const height = containerRef.current ? containerRef.current.offsetHeight : 0;
+  React.useEffect(
+    () => {
+      eventsHeightsCache.setEventHeight(event, height);
+    },
+    [event, eventsHeightsCache, height]
+  );
 
-  _onEventUpdated = () => {
-    this.forceUpdate();
+  const _onEventUpdated = () => {
```

Comment:
Could be replace by forceUpdate? This would give at least a stable function between renders, which is always a win (it's one less prop that is changing at every render for the children :))